src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -341,7 +437,7 @@ protected virtual DefaultMetadataDetails CreateParameterDetails(ModelMetadataIde
                 ModelAttributes.GetAttributesForParameter(key.ParameterInfo, key.ModelType));
         }
 
-        private class TypeCache : ConcurrentDictionary<ModelMetadataIdentity, ModelMetadataCacheEntry>
+        private class ModelMetadataCache : ConcurrentDictionary<ModelMetadataIdentity, ModelMetadataCacheEntry>
```

Comment:
Why the churn❔

Reply:
This is a bit of a misnomer - it's not longer a cache for types, but also constructors, parameters and properties. We never changed this once it evolved past being a cache for types.